src/runtime/scheduler/scheduler.rs

```diff
@@ -55,40 +55,27 @@ impl Scheduler {
         self.groups.insert(TaskGroup::default()).into()
     }
 
-    fn get_mut_group(&mut self, id: SchedulerId) -> Option<&mut TaskGroup> {
+    fn get_group_mut(&mut self, id: SchedulerId) -> Option<&mut TaskGroup> {
         self.groups.get_mut(id.into())
     }
 
-    /// The parent id can either be the id of the group or another task in the same group.
-    pub fn insert_and_poll_task<T: Task>(&mut self, group_id: SchedulerId, task: T) -> Option<InsertResult> {
-        let group: &mut TaskGroup = self.get_mut_group(group_id)?;
-        group.insert_and_poll_task(Box::new(task))
+    pub fn insert<T: Task>(&mut self, group_id: SchedulerId, task: T) -> Option<InsertResult> {
+        self.get_group_mut(group_id)?.insert(Box::new(task))
     }
 
-    pub fn get_mut_task(&mut self, group_id: SchedulerId, task_id: SchedulerId) -> Option<Pin<&mut Box<dyn Task>>> {
-        let group: &mut TaskGroup = self.get_mut_group(group_id)?;
-        group.get_mut_task(task_id.into())
+    pub fn get_task_mut(&mut self, group_id: SchedulerId, task_id: SchedulerId) -> Option<Pin<&mut Box<dyn Task>>> {
+        self.get_group_mut(group_id)?.get_task_mut(task_id.into())
     }
 
-    /// Polls all ready tasks in this group until there are no runnable ones. Do not use this function on coroutines
-    /// that use poll_yield, unless max_iterations is set.
-    pub fn poll_group_until_unrunnable(
-        &mut self,
-        group_id: SchedulerId,
-        max_iterations: Option<usize>,
-    ) -> Vec<Box<dyn Task>> {
-        // Expect is safe here because something has really gone wrong if we are polling a group that doesn't exist.
-        let group: &mut TaskGroup = self.get_mut_group(group_id).expect("group being polled doesn't exist");
-
-        // Keep polling the group and checking for runnable tasks until there are none left.
-        group.poll_group(max_iterations, true)
+    /// Do not use this function on coroutines that use poll_yield, unless limit is set.
+    pub fn poll_group(&mut self, id: SchedulerId, limit: Option<usize>) -> Vec<Box<dyn Task>> {
```

Comment:
we should probably settle on either poll or run as the verb in all of our variable names